Lib/mimetypes.py

```diff
@@ -141,25 +141,22 @@ def guess_type(self, url, strict=True):
                 type = 'text/plain'
             return type, None           # never compressed, so encoding is None
         base, ext = posixpath.splitext(url)
-        while ext in self.suffix_map:
-            base, ext = posixpath.splitext(base + self.suffix_map[ext])
+        while (ext_lower := ext.lower()) in self.suffix_map:
+            base, ext = posixpath.splitext(base + self.suffix_map[ext_lower])
         if ext in self.encodings_map:
```

Comment:
From the issue it looks like it's intentional that we don't casefold the encodings_map lookup, but it's not obvious to a reader of the code. Could you add a comment explaining why we shouldn't lower here?

Reply:
Encodings map is case sensitive hence it should not be lowered, I added a comment.